src/liboslcomp/artic.cpp

```diff
@@ -223,4 +234,109 @@ ArticSource::integrate_temp_source(ArticSource&& articSource)
     this->save_temp(std::move(articSource.m_code));
 }
 
+void ArticTranspiler::dispatch_node(ASTNode::ref n)
+{
+    auto node = n.get();
+    switch (node->nodetype()) {
+    case ASTNode::unknown_node:
+        NOT_IMPLEMENTED;
+        break;
+    case ASTNode::shader_declaration_node:
```

Comment:
there is a nice macro trick we use in thorin to avoid this kind of thing, i can explain it to you in a call at some point